main/src/ca/uwaterloo/flix/language/phase/Regions.scala

```diff
@@ -203,8 +203,8 @@ object Regions {
     case Expr.Do(_, exps, tpe, _, loc) =>
       exps.flatMap(visitExp) ++ checkType(tpe, loc)
 
-    case Expr.InvokeMethod2(exp, _, exps, tpe, _, loc) =>
-      exps.flatMap(visitExp) ++ visitExp(exp) ++ checkType(tpe, loc)
+    case Expr.InvokeMethod2(exp, _, exps, mtpe, tpe, _, loc) =>
+      exps.flatMap(visitExp) ++ visitExp(exp) ++ checkType(mtpe, loc) ++ checkType(tpe, loc) // do we check mtpe?
```

Comment:
@mlutze In my current plan, InvokeMethod2 has already been erased here. I think that is OK, no?

Reply:
Seems ok. We keep all the subexpressions and the type, right? So only `mtpe` is lost and if that has a region in it then I think we have bigger problems.